src/SwapRouter.sol

```diff
@@ -0,0 +1,210 @@
+    }
+
+    /// @dev Direction => True => CRV -> CVXCRV
+    /// @dev Direction => False =>  CVXCRV -> CRV
+    function swapOnCRVCVXCRVPool(
+        bool direction,
+        uint256 amount,
+        address recipient
+    ) external override returns (uint256 amountOut) {
+        IERC20Metadata swapToken = direction ? CRV : CVXCRV;
+        IERC20Metadata recievedToken = direction ? CVXCRV : CRV;
+
+        require(
+            amount > 0 && amount <= swapToken.balanceOf(msg.sender),
+            "SwapRouter :: amount"
+        );
+        require(recipient != address(0), "SwapRouter :: recipient");
+
+        swapToken.safeTransferFrom(msg.sender, address(this), amount);
+        crvcvxcrvPool.exchange(
+            int8(direction ? 0 : 1),
+            int8(direction ? 1 : 0),
+            amount,
+            0,
+            address(this)
+        );
+
+        amountOut = recievedToken.balanceOf(address(this));
+
+        swapToken.safeTransfer(recipient, swapToken.balanceOf(address(this)));
+        recievedToken.safeTransfer(recipient, amountOut);
+    }
+
+    function burn3CRVForUSDC(uint256 amount, address recipient)
+        external
+        override
+        returns (uint256 amountOut)
+    {
+        require(
+            amount > 0 && amount <= _3CRV.balanceOf(msg.sender),
+            "SwapRouter :: amount"
+        );
+        require(recipient != address(0), "SwapRouter :: recipient");
+
+        _3CRV.safeTransferFrom(msg.sender, address(this), amount);
+        /// @dev i = 1 --> USDC
+        _3crvPool.remove_liquidity_one_coin(amount, 1, 0);
+
+        amountOut = USDC.balanceOf(address(this));
+
+        _3CRV.safeTransfer(recipient, amount);
+        USDC.safeTransfer(recipient, amountOut);
+    }
+
+    function getTokenPriceInUSD(address token) public view returns (uint256) {
+        require(
+            token == address(CRV) || token == address(CVX),
+            "SwapRouter :: token"
+        );
+
+        (, int256 answer, , , ) = (token == address(CRV) ? CRVUSD : CVXUSD)
+            .latestRoundData();
+
+        return (uint256(answer) / uint256(CRVUSD.decimals())) * USDC.decimals();
+    }
+
+    function _swapTokens(
+        IERC20Metadata token0,
+        IERC20Metadata token1,
+        uint256 amount,
+        uint256 minReturn,
+        uint256 slippage,
+        bytes memory data
+    ) internal {
+        SwapDescription memory desc = SwapDescription({
+            srcToken: IERC20(address(token0)),
+            dstToken: IERC20(address(token1)),
+            srcReceiver: payable(aggregationExecutor),
+            dstReceiver: payable(address(this)),
+            amount: amount,
+            minReturnAmount: (minReturn * (100 - slippage)) / 100,
+            flags: 0,
+            permit: "0x"
+        });
+
+        oneInchRouter.swap(
+            IAggregationExecutor(aggregationExecutor),
+            desc,
+            data
+        );
+    }
+
+    function sweep() external {
```

Comment:
sweep takes token `address` and sweep only that token not all. Mainly used for transferring tokens that are not handled by strategy